test/dt_arith.c

```diff
@@ -601,19 +608,34 @@ static int
 test_particular_fp_integer(void)
 {
     hid_t          dxpl_id;
-    bool           flag;
+    int            flag;
     double         src_d = (double)SCHAR_MAX;
     signed char    dst_c;
-    unsigned char *buf1 = NULL, *buf2 = NULL;
-    unsigned char *saved_buf1 = NULL, *saved_buf2 = NULL;
-    size_t         src_size1, src_size2;
-    size_t         dst_size1, dst_size2;
-    float          src_f = (float)INT_MAX;
-    int            dst_i;
-    int            fill_value = 13;
+    unsigned char *buf1       = NULL;
+    unsigned char *saved_buf1 = NULL;
+    size_t         src_size1;
+    size_t         dst_size1;
     int            endian; /*endianness            */
     unsigned int   fails_this_test = 0;
     size_t         j;
+#ifdef H5_WANT_DCONV_EXCEPTION
```

Comment:
The changes in this function add `_Float16` testing and also skip running this part of the test when `H5_WANT_DCONV_EXCEPTION` is not defined, as there was otherwise a testing failure in that case since the library wouldn't raise the expected floating-point conversion exception.